static/js/app.js

```diff
@@ -288,14 +299,28 @@ const App = {
     const extensionId = context.params.extensionId;
 
     if (this.extensions.hasOwnProperty(extensionId)) {
-      this.extensions[extensionId].show(context);
-      this.selectView(
-        `extension-${Utils.escapeHtmlForIdClass(extensionId)}-view`
-      );
-      this.displayedExtension = extensionId;
+      const extension = this.extensions[extensionId];
+      // wait until the extensions is loaded before showing it
+      extension.load()
+        .then(() => {
+          extension.show(context);
+          this.extensions[extensionId].show(context);
+          this.selectView(
+            `extension-${Utils.escapeHtmlForIdClass(extensionId)}-view`
+          );
+          this.displayedExtension = extensionId;
+        });
```

Comment:
You should add a `catch()` that will just redirect to `/things`.

Reply:
I added the `catch` accordingly.